data/data/azurestack/cluster/master/master.tf

```diff
@@ -54,7 +54,7 @@ resource "azurestack_virtual_machine" "master" {
   storage_os_disk {
     name              = "${var.cluster_id}-master-${count.index}_OSDisk" # os disk name needs to match cluster-api convention
     create_option     = "FromImage"
-    disk_size_gb      = min(var.os_volume_size, 1023)
+    disk_size_gb      = max(var.os_volume_size, 1023)
```

Comment:
Hm I think my comment about max vs min in the jira card was mistaken. `min` seems correct here. If we set this to `max` and `1023` is the maximum value according to the bug, then the disk size will always be set to 1023. The default value of 128 would effectively be ignored.
https://developer.hashicorp.com/terraform/language/functions/max

Reply:
Changing this to 
```
max(var.os_volume_size, 128)
```
would work, but we would need to make sure there is validation to cap the disk size at 1023, at which point it would seem to make more sense to just add validation to make sure disk size is between 128 and 1023

Reply:
How about current change? Not sure if it would even work